src/qibolab/platforms/utils.py

```diff
@@ -0,0 +1,93 @@
+from types import SimpleNamespace
+
+
+class Channel:
+    """Representation of physical wire connection (channel).
+
+    Name is used as a unique identifier for channels. If a channel
+    with an existing name is recreated, it will refer to the existing object.
+    Channel objects are instantiated by :class:`qibolab.platforms.platform.Platform`,
+    but their attributes are modified and used by instrument designs.
```

Comment:
I understand this is usually working fine, and perfectly allowed by Python, but it is not intuitive at all.

Instead of working out your own variable system and so on, prefer using Python one. Globals are always dangerous (especially in concurrent environments, but they might be confusing in general), and usually bad design.
Consider the case in which the same application is spawning two platforms: you have to make sure they do not share names, otherwise they will overwrite each other.

Instead, prefer creating a `Channels` class, if you want to have a container, that can be a wrapper around a dictionary (if you want to address channels by naming with strings) with a method to instantiate new channels, like:
```py
channels = Channels()
abc: Channel = channels.new("abc")
abc.local_oscillator = abclo
abc.lo_frequency = abcfreq
channels.new("xyz")
channels["xyz"].local_oscillator = xyzlo
```